### PR TITLE
Chore: setup dependency injection

### PR DESCRIPTION
**Pull request by ahmdyaasiin**
- Initialize Project
   [+] Initialize directory structure for event
   [+] Add .gitignore for .idea directory and .env files
- Documentation
   [+] Add OpenAPI file based on new design
- Database
   [+] Add migration files for each service 
   [+] Convert database schema into go file
- Dependency Injection
   [+] Add Uber Dig as dependency injection
- gRPC Service
   [+] Add proto file for event
   [+] Generate protobuf from event.proto
   [+] Implement gRPC server with Uber Dig

### CODE REVIEW

**Comment by mirzahilmi** [comment]
60+ files changed, what a massive PR

File: challenge-service/internal/domain/challenge.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package domain
+
+import "time"
+
+type Challenge struct {
+	ID             int       `db:"id"`
```
Comment (mirzahilmi) [inline challenge-service/internal/domain/challenge.go:6]: is the datatype correct here? in the migration file you provided its type of char(26)

File: challenge-service/internal/domain/challenge.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	EndDate        time.Time `db:"end_date"`
+	EndTime        time.Time `db:"end_time"`
+	CertificateURL string    `db:"certificate_url"`
+	PhotoURLs      []string  `db:"photo_urls"`
+	CreatedAt      time.Time `db:"created_at"`
+	UpdatedAt      time.Time `db:"updated_at"`
+}
+
+type ChallengeSubmission struct {
+	ID                      int       `db:"id"`
```
Comment (mirzahilmi) [inline challenge-service/internal/domain/challenge.go:22]: again, is the datatype correct here?

File: recruitment-service/internal/domain/recruitment.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
+package domain
+
+import "time"
+
+type Recruitment struct {
+	ID          int       `db:"id"`
```
Comment (mirzahilmi) [inline recruitment-service/internal/domain/recruitment.go:6]: same comment, is this the correct datatype?

File: recruitment-service/internal/domain/recruitment.go [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+	StartTime   time.Time `db:"start_time"`
+	EndDate     time.Time `db:"end_date"`
+	EndTime     time.Time `db:"end_time"`
+	PhotoURLs   []string  `db:"photo_urls"`
+	CreatedAt   time.Time `db:"created_at"`
+	UpdatedAt   time.Time `db:"updated_at"`
+}
+
+type RecruitmentRegistrant struct {
+	ID                      int       `db:"id"`
```
Comment (mirzahilmi) [inline recruitment-service/internal/domain/recruitment.go:20]: same comment, is this the correct datatype?

File: workshop-service/cmd/app/main.go [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+	"github.com/bccfilkom/backend/workshop-service/internal/bootstrap/server"
+	"github.com/bccfilkom/backend/workshop-service/internal/di"
+	"go.uber.org/dig"
+	"log"
+)
+
+var container = dig.New()
+
+func main() {
+	//
```
Comment (mirzahilmi) [inline workshop-service/cmd/app/main.go:13]: what's up with this empty comment :) please dont include unnecessary loc

File: workshop-service/internal/bootstrap/config/env.go [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+	env := os.Getenv("ENV")
+
+	if err != nil && env == "" {
+		log.Panicf("Failed to load .env file, err: %v\n", err)
+	}
+
+	conf := &AppConfig{
+		AppPort: os.Getenv("APP_PORT"),
+
+		// add more
```
Comment (mirzahilmi) [inline workshop-service/internal/bootstrap/config/env.go:24]: no unnecessary comment

File: workshop-service/internal/infrastructure/grpc/event.proto
Comment (mirzahilmi) [inline workshop-service/internal/infrastructure/grpc/event.proto:1]: too large, can you break it up into multiple files? see [here](https://protobuf.dev/programming-guides/proto2/#importing) for the example of multiple proto files

File: workshop-service/cmd/app/main.go (hunk 5 above)
Comment (ahmdyaasiin) [reply workshop-service/cmd/app/main.go]: i've deleted the unnecessary comment, please check the last commit

File: workshop-service/internal/bootstrap/config/env.go (hunk 6 above)
Comment (ahmdyaasiin) [reply workshop-service/internal/bootstrap/config/env.go]: i've deleted this unecessary comment as well, please check the last commit

**Pull request closed without merging**